lib/core/include/connection_pool.hpp

```diff
@@ -57,6 +57,8 @@ namespace irods
 
         connection_proxy get_connection();
 
+        static std::shared_ptr<connection_pool> make_one();
```

Comment:
This should probably be moved outside of the class as it doesn't depend on anything within it. This would allow the name to be more descriptive as well as accept an argument for specifying the size of the pool.
```c++
std::shared_ptr<connection_pool> make_connection_pool(int size = 1);
```

Reply:
I like it